code/datums/extensions/cafe_prop.dm

```diff
@@ -0,0 +1,57 @@
+/datum/extension/cafe_prop
+	base_type = /datum/extension/cafe_prop
+	expected_type = /obj/item/chems
+
+/datum/extension/cafe_prop/proc/update_prop(var/obj/item/prop)
+	if(!istype(prop))
+		return
+
+	. = initial(prop.name)
+	if(prop.reagents?.total_volume > 0)
+		. = "[.] of"
+		var/decl/reagent/primary_reagent = prop.reagents.get_primary_reagent_decl()
+		if(primary_reagent.type != /decl/reagent/drink/ice && prop.reagents.has_reagent(/decl/reagent/drink/ice))
+			. = "[.] iced"
+		
+		if(istype(primary_reagent, /decl/reagent/drink/coffee))
+			
+			if(prop.reagents.has_reagent(/decl/reagent/nutriment/mint))
+				. = "[.] mint"
+			if(prop.reagents.has_reagent(/decl/reagent/drink/syrup_caramel))
+				. = "[.] caramel"
+			if(prop.reagents.has_reagent(/decl/reagent/drink/syrup_vanilla))
+				. = "[.] vanilla"
+			if(prop.reagents.has_reagent(/decl/reagent/drink/syrup_pumpkin))
+				. = "[.] pumpkin spice"
+			if(prop.reagents.has_reagent(/decl/reagent/drink/syrup_chocolate))
+				. = "[.] mocha"
+
+			var/milk =  REAGENT_VOLUME(prop.reagents, /decl/reagent/drink/milk)
+			var/soy =   REAGENT_VOLUME(prop.reagents, /decl/reagent/drink/milk/soymilk)
+			var/cream = REAGENT_VOLUME(prop.reagents, /decl/reagent/drink/milk/cream)
+			var/chai =  REAGENT_VOLUME(prop.reagents, /decl/reagent/drink/tea/chai) ? " chai " : " "
+
+			if(!soy && !milk && !cream)
+				. = "[.][chai]flavoured coffee"
+			else if(milk > soy || cream > soy)
```

Comment:
What about the case where milk and cream are greater than soy when combined? Maybe use `(milk + cream) > soy` or something.